tt_eager/tt_dnn/op_library/bmm/bmm_op.hpp

```diff
@@ -351,37 +350,21 @@ inline Tensor matmul(
         output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor_a, input_tensor_b}))};
     }
 
-    if (!needs_autoformat) {
-        operation::launch_op(
-                [program_config, mem_config, output_dtype, compute_kernel_config, untilize_out, user_core_coord, user_fused_activation, input_b_is_batched] (const std::vector<Tensor>& input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors, const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
-            const auto& input_tensor_a = input_tensors.at(0);
-            const auto& input_tensor_b = input_tensors.at(1);
-            auto arch = input_tensor_a.device()->arch();
-            const bool has_user_grid = user_core_coord.has_value();
-            const bool has_program_config = program_config.has_value();
-            const auto increase_fidelity = !has_program_config && !has_user_grid;
-            auto math_fidelity = increase_fidelity ? MathFidelity::HiFi2 : MathFidelity::LoFi;
-            auto kernel_config_val = init_device_compute_kernel_config(arch, compute_kernel_config, math_fidelity);
-            bool broadcast_batch = get_broadcast_batch(input_tensor_a, input_tensor_b, program_config);
-            TT_FATAL(!(has_user_grid && has_program_config), "Cannot use both user core grid/coordinates and a program config");
-            return operation::run(Matmul{program_config, broadcast_batch, mem_config, output_dtype.value_or(input_tensor_a.get_dtype()), kernel_config_val, untilize_out, user_core_coord, user_fused_activation, input_b_is_batched}, {input_tensor_a, input_tensor_b}, optional_input_tensors);
-        },
-        {input_tensor_a, input_tensor_b}, output_tensors, optional_input_tensors);
-    } else {
-        operation::launch_with_autoformat(
-                [program_config, mem_config, output_dtype, compute_kernel_config, untilize_out, user_core_coord, user_fused_activation, input_b_is_batched] (const std::vector<Tensor>& input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors, const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
```

Comment:
dumb question, but what does autoformat do exactly?

Reply:
It does a bunch of things, automatically preparing the tensors to be used on a device. The main ones are that it
- pads tensors appropriately and
- moves them onto the device if they are not on device

outer() had [1,1,M,1] x [1,1,1,N] tensors. The device tensors cannot support shapes that end in 1.
I finally moved enough logic into outer() with Akhmed's help last week to not need launch_with_autoformat in matmul.

Reply:
thanks